main.js

```diff
@@ -2233,6 +2234,20 @@ function processMessage(msg) {
             })(msg);
             break;
         }
+
+        case 'rebuildAdapter':
+            (function (msg) {
+                if (!installQueue.find(entry => entry && entry.id && entry.id === msg.message.id)) {
```

Comment:
The check in the predicate function seems unnecessary. Is there ever a case where `installQueue` contains `undefined` or `null`? Also, `msg.message.id` shouldn't be undefined, so the following should be enough:

```js
if (!installQueue.some(entry => entry.id === msg.message.id)) {
```
`.some` has been supported for a while and just returns true/false instead of the found object.

Reply:
For the "predicate function" ... here I did like the others commands. I add a TODO to refacor this. Other change will be done. good idea

Reply:
Ok, I removed predacate fuctions here and in some other places ... onyl one I left because unsure (checkLogging)